### PR TITLE
Add RWMutex lock for clients map to prevent concurrent writes

### LINKED ISSUE #42: Collector concurrent map writes
Unit tests failed here https://github.com/vmware/go-ipfix/runs/1163485465 due to concurrent map writes on client handlers.

### PR DESCRIPTION
**Pull request by zyiou**
fixes #42

### CODE REVIEW

File: pkg/collector/process.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ type collectingProcess struct {
 	messages []*entities.Message
 	// maps each client to its client handler (required channels)
 	clients map[string]*clientHandler
+	// clientsLock allows multiple readers or one writer to clients map at the same time
```
Comment (srikartati) [inline pkg/collector/process.go:49]: to access  clients map is missing

**Review by srikartati — DISMISSED** [review]
LGTM except for nit

File: pkg/collector/process.go [hunk 2]
```diff
@@ -46,6 +46,8 @@ type collectingProcess struct {
 	messages []*entities.Message
 	// maps each client to its client handler (required channels)
```
Comment (srikartati) [inline pkg/collector/process.go:47]: Not related to this PR. I feel it is better to have uniformity/convention when adding comments to the fields, i.e., starting the comment with the field name. what do you say?

**Review by srikartati — COMMENTED** [review]
A nit about comments. It's better to follow a convention.

File: pkg/collector/process.go (hunk 2 above)
Comment (zyiou) [reply pkg/collector/process.go:47]: Makes sense. I will create a PR to uniform all the comments. Thanks!

**Review by srikartati — APPROVED** [review]

**Pull request merged**